docs/examples/crowd-funding.rst

```diff
@@ -0,0 +1,153 @@
+********************
+Secure Crowd Funding
+********************
+
+In this example We will see how to create an crowdfunding smart contract which will quite secure by moving the power of campaign's admin to those who donated money.
+
+Let's first design all the things:
+
+1. The Admin will start a campaign for crowdfunding with a specefic monetary goal and deadline.
+
+2. Contributors will contribute to that project by sending eth.
+
+3. The admin has to create a spending request to spend money for the campaign.
+
+4. Once the spending request was created,the contributors can start voting for that spending request.
+
+5. If more than 50% of the total contributors voted for the request,then the admin would have the permission to spend the amount specefic in the spending request.
+
+6. The powe is moved from the campaign's admin to those that donated money.
+
+7.The contributors can request a refund if the monetary goal was not reached within the deadline. 
+
+The full contract
+-----------------
+
+.. code-block:: solidity
+    :force:
+
+    // SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.5;
+ 
+contract CrowdFunding {
+    mapping(address => uint) public contributors;
+    address public admin;
+    uint public noOfContributors;
+    uint public minimumContribution;
+    uint public deadline; //timestamp
```

Comment:
Maybe a good choice for user defined value types?

Something like `type Timestamp is uint256;`, then `Timestamp public deadline`. Maybe this can also be immutable? If not, usually timestamps are shorter than `256` bits so that it can be packed.

https://blog.soliditylang.org/2021/09/27/user-defined-value-types/

Reply:
I actually don't get this , that what is the benefit of this using here.

Reply:
Good question. Better abstraction?